test/vagrant-service-manager/installer_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+require "vagrant/util/downloader"
+
+module VagrantPlugins
+  module ServiceManager
+    describe Installer do
+      before do
+        @type = :docker
+        @machine = fake_machine
+        @env = @machine.env
+        @options = { box_version: 'adb', '--cli-version' => '1.11.0' }
+
+        # set bin and tmp directories
+        @base_test_path = "#{@env.data_dir}/service-manager/test"
+        ServiceManager.set_bin_dir("#{@base_test_path}/bin")
+        ServiceManager.set_temp_dir("#{@base_test_path}/tmp")
+        @bin_dir = ServiceManager.bin_dir
+        @temp_dir = ServiceManager.temp_dir
+
+        # Installer instance
+        @installer = Installer.new(@type, @machine, @env, @options)
+      end
+
+      after do
+        FileUtils.rmtree(@bin_dir) if File.directory? @bin_dir
+        FileUtils.rmtree(@temp_dir) if File.directory? @temp_dir
+      end
+
+      it 'should set default values properly' do
+        @installer.instance_variable_get('@binary_exists').must_equal true
+        @installer.instance_variable_get('@url').must_equal ''
```

Comment:
@hferentschik Ya I am having this doubt too. Like there is @path variable which I am testing but I couldn't test `--cli-version` since I didn't make its instance variable.

Hope moving these to handler or base of handlers will solve all these issues